app/routes.py

```diff
@@ -1 +1,280 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, abort, make_response
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from sqlalchemy import desc
+from datetime import datetime
+import requests
+import os
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_model(cls, model_id): 
+    try: 
+        model_id = int(model_id)
+    except: 
+        abort(make_response({"message":f"{model_id} invalid type ({type(model_id)})"}, 400))
+        # abort(make_response({"message":f"Invalid data"}, 400))
+        
+    model = cls.query.get(model_id) 
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__.lower()} {model_id} not found"}, 404))
+
+    return model
+
+
+# Create a Task: Valid Task with 'null' 'completed at'
+@task_bp.route("", methods=["POST"])
+def handle_tasks(): 
+    # handle the HTTP request body - pasrses JSON body into a Python dict
+    request_body = request.get_json()
+
+    # create a new task instance from the request
+    if len(request_body) < 2:
+        return {"details": "Invalid data"}, 400
+    else: 
+        new_task = Task.from_dict(request_body)
+
+    # database collects new changes - adding new_task as a record
+    db.session.add(new_task)
```

Comment:
Great job using `db.session.<method-name>`. Session gives us access to the follow:

- `db` is our app’s instance of SQLAlchemy.
- `session` represents our active database connection.
- By referencing `db.session` we can use SQLAlchemy’s methods to perform tasks like:
    - committing a change to a model
    - storing a new record of a model
    - deleting a record.
- By referencing `db.session.add()` you are able to use the SQLAlchemy method to store a new record of the task model